### PR TITLE
Backport 13066, 13448 and fix `commit-script-check.sh` to make shellcheck 0.7.2 happy

### PR DESCRIPTION
**Pull request by UdjinM6**
Fixes `SC2268: Avoid x-prefix in comparisons as it no longer serves a purpose.`

https://github.com/koalaman/shellcheck/wiki/SC2268

### CODE REVIEW

File: contrib/verify-commits/allow-incorrect-sha512-commits [hunk 1]
```diff
@@ -0,0 +1,2 @@
+f8feaa4636260b599294c7285bcf1c8b7737f74e
+8040ae6fc576e9504186f2ae3ff2c8125de1095c
```
Comment (PastaPastaPasta) [inline contrib/verify-commits/allow-incorrect-sha512-commits:2]: We shouldn't have these?

File: contrib/verify-commits/allow-unclean-merge-commits [hunk 2]
```diff
@@ -0,0 +1,4 @@
+6052d509105790a26b3ad5df43dd61e7f1b24a12
+3798e5de334c3deb5f71302b782f6b8fbd5087f1
+326ffed09bfcc209a2efd6a2ebc69edf6bd200b5
+97d83739db0631be5d4ba86af3616014652c00ec
```
Comment (PastaPastaPasta) [inline contrib/verify-commits/allow-unclean-merge-commits:4]: same

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by xdustinface — APPROVED** [review]
utACK

**Pull request merged**